miner/miner.go

```diff
@@ -2,29 +2,98 @@ package miner
 
 import (
 	"math"
+	"sync"
 	"time"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/ubclaunchpad/cumulus/blockchain"
 	"github.com/ubclaunchpad/cumulus/consensus"
 )
 
+// currentlyMining is a flag to control the miner.
+var currentlyMining bool
+
+// currentlyMiningLock is a read/write lock to change the Mining flag.
+var currentlyMiningLock sync.RWMutex
+
+const (
+	// MiningSuccessful is returned when the miner mines a block.
+	MiningSuccessful = iota
+	// MiningNeverStarted is returned when the block header is invalid.
+	MiningNeverStarted = iota
+	// MiningHalted is returned when the app halts the miner.
+	MiningHalted = iota
+)
+
+// MiningResult contains the result of the mining operation.
+type MiningResult struct {
```

Comment:
Why have this be a type when we can just return `(complete bool, info int)` from `Mine()`? Do we use MiningResult outside this file?